### PR TITLE
feat: Lerna Integration

### PR DESCRIPTION
**Pull request by vkbansal**
You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

File: packages/uicore/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "baseUrl": "./src",
+    "outDir": "./dist",
+    "alwaysStrict": true,
+    "module": "esnext",
+    "lib": ["esnext", "dom"],
+    "moduleResolution": "node",
+    "target": "ES6",
```
Comment (knagurski) [inline packages/uicore/tsconfig.json:10]: Might be worth upping this to something more modern while we're here 🤔

File: scripts/publish.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+echo always-auth=true >> .npmrc
+BASE_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+
+CHANGED=$(npx lerna changed)
+
+if [[ "$(echo $CHANGED | wc -l)" -gt 0 ]]
+then
+    echo $CHANGED
+    yarn build
+    yarn release minor --yes --ignore-scripts
```
Comment (knagurski) [inline scripts/publish.sh:30]: Is there anyway we can allow this to be overridden when we know it'll be a major or just a patch? Probably a wider conversation to be had around that.

File: packages/uicore/tsconfig.json (hunk 1 above)
Comment (vkbansal) [reply packages/uicore/tsconfig.json:10]: I'll take this up for vite migration

File: scripts/publish.sh (hunk 2 above)
Comment (vkbansal) [reply scripts/publish.sh:30]: Yeah let's discuss this in tonight's meeting

**Comment by vkbansal** [comment]
retrigger titlecheck

**Comment by vkbansal** [comment]
retrigger prettier

**Comment by vkbansal** [comment]
retrigger typecheck

**Comment by vkbansal** [comment]
retrigger typecheck

**Comment by vkbansal** [comment]
retrigger eslint

**Comment by vkbansal** [comment]
retrigger stylelint

**Comment by vkbansal** [comment]
retrigger stylelint

**Review by kapilkumawat86 — DISMISSED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**